app/code/Magento/Catalog/Model/ResourceModel/Product/Indexer/Eav/AbstractEav.php

```diff
@@ -148,6 +148,21 @@ public function reindexAttribute($attributeId, $isIndexable = true)
     abstract protected function _prepareIndex($entityIds = null, $attributeId = null);
 
     /**
+     * Prepare product visibility attribute value index
+     *
+     * @param array $entityIds      the entity ids limitation
+     * @return $this
+     */
+    protected function _prepareVisibilityIndex($entityIds = null)
+    {
+        $attribute = $this->_eavConfig->getAttribute(\Magento\Catalog\Model\Product::ENTITY, 'visibility');
```

Comment:
It's better to use constant from Public Service Contract API 
\Magento\Catalog\Api\Data\ProductAttributeInterface::ENTITY_TYPE_CODE